tool/tsh/kubectl.go

```diff
@@ -134,9 +140,14 @@ func runKubectlCode(args []string) {
 }
 
 func runKubectlAndCollectRun(cf *CLIConf, args []string) error {
+	closeFn, err := maybeStartKubeLocalProxy(cf, withKubectlArgs(args), withKubeconfigEnvUpdate())
```

Comment:
This only needs to happen in `runKubectlReexec` and you set the env variable in the `cmd.Env`

Reply:
Moved to `runKubectlReexec`, and now setting the new location in cmd.Env.

For `tsh kube exec`, changed to
```
    if overwriteKubeConfigLocation != "" {                                
        kubeConfigFlags.KubeConfig = pointer.String(overwriteKubeConfigLocation)                                
    }  
```